rest/rest.go

```diff
@@ -103,5 +183,61 @@
 
 	log.Printf("Starting REST gateway on :%d", httpPort)
 
-	return srv.ListenAndServe()
+	sock, err = net.Listen("tcp", srv.Addr)
+	if err != nil {
```

Comment:
Add some error message?

Reply:
I wanted to emulate the behaviour of the previous `ListenAndServe` which also returns the error directly. The only reason for the change is that I need to access the `sock`, to get the actual port used for unit tests. The unit tests are started with port `0`, which is the "next available one".